### DISCUSSION TITLE
Metaclass arguments vs. `model_config` attribute

### DISCUSSION THREAD

**Discussion by basnijholt** [pydantic/pydantic#8831] [Question]
Both of the following examples seem to do the same:
```python
from pydantic import BaseModel, ConfigDict

class Foo(BaseModel):
    a: int
    model_config: ConfigDict = ConfigDict(frozen=True)

class Bar(Foo, frozen=True):
    a: int
```

`Foo` uses `model_config` and `Bar` uses metaclass arguments. Personally, I prefer the latter approach, however, nowhere in the documentation (v2) I can find any reference to using metaclass arguments to set config attributes.

I would like to hear from the pydantic core devs what your recommendation would be. Whether or not there are any disadvantages to using metaclass arguments to set the config?

**Comment by basnijholt** [comment]
@samuelcolvin or @sydney-runkle? I ran into this question again today.

**Comment by Viicos** [comment]
You can read more about it [here](https://docs.pydantic.dev/latest/integrations/visual_studio_code/#config-in-class-arguments) (albeit this page is a bit outdated). The benefit is you get type checking support for specific arguments such as `frozen`. I agree we should better document this.

**Reply by basnijholt** [reply]
The reason I came back to this issue was that mypy actually complains when using metaclass arguments.

**Reply by Viicos** [reply]
What kind of error do you get? What's your mypy version?
